packages/main/src/server/cache-manager.ts

```diff
@@ -0,0 +1,169 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CachedContent, Content, Part, RequestOptions, Tool, ToolConfig } from "../../types";
+import {
+  CachedContentUrl,
+  FilesRequestUrl,
+  getHeaders,
+  makeServerRequest,
+} from "./request";
+import {
+  CachedContentCreateParams,
+  CachedContentUpdateParams,
+  ListCacheResponse,
+  ListParams,
+} from "../../types/server";
+import { RpcTask } from "./constants";
+import { GoogleGenerativeAIError } from "../errors";
+
+/**
+ * Class for managing GoogleAI file uploads.
+ * @public
+ */
+export class GoogleAICacheManager {
+  model: string;
+  ttl?: string;
+  tools?: Tool[];
+  toolConfig?: ToolConfig;
+  systemInstruction?: string | Part | Content;
+
+  constructor(
+    public apiKey: string,
+    private _requestOptions: RequestOptions,
+  ) {}
+
+  /**
+   * Upload a new content cache
```

Comment:
Should be create a new instance of CachedContent, not file upload, and maybe mention that it hits the service..?

Reply:
I think "upload" is pretty clear that it hits the service? It doesn't create any instance. I notice actually all the variables for headers in this class are called `uploadHeaders` for some reason so I changed them to `headers`. Seems to be a problem with GoogleAIFileManager too but don't want to mess with too many files at once. It's not public facing anyway.